### PR TITLE
Tractogram scripts part1

### PR DESCRIPTION
**Pull request by EmmaRenauld**
# Quick description

Verified the four first tractogram scripts.
1. `scil_tractogram_apply_transform`: 
    - Fixed difference between the doc's explanation and the script's management of option --remove_invalid.
2. `scil_tractogram_apply_transform_to_hdf5`:
   - Made the same choices of management of invalid as in script 1.
   - ENCAPSULATED HDF5 MANAGEMENT IN SCILPY.IO.HDF5.  (Many of these methods are not used here but will be in the next PR. I just wanted to make sure right now that everything fitted correctly).
3. `scil_tractogram_assign_uniform_color`:
   - Moved loading of the json file out of the loop.
   - Better explanation in parser
   - Better verifications in main
   - Encapsulated hexadecimal_to_RGB in viz.utils.
4. `scil_tractogram_cut_streamlines`:
   - Now the data_per_streamline is kept.

## Type of change

Check the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Provide data, screenshots, command line to test (if relevant)

...

# Checklist

- [ ] My code follows the style guidelines of this project (run [autopep8](https://pypi.org/project/autopep8/))
- [ ] I added relevant citations to scripts, modules and functions docstrings and descriptions
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I moved all functions from the script file (except the argparser and main) to scilpy modules
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by AntoineTheb — CHANGES_REQUESTED** [review]
Minor comments about the text, otherwise LGTM.

File: scilpy/tractograms/streamline_and_mask_operations.py [hunk 1]
```diff
@@ -128,7 +128,8 @@ def cut_outside_of_mask_streamlines(sft, binary_mask, min_len=0):
     new_streamlines = _cut_streamlines_with_masks(
         streamlines, binary_mask, binary_mask)
 
-    new_sft = StatefulTractogram.from_sft(new_streamlines, sft)
+    new_sft = StatefulTractogram.from_sft(
+        new_streamlines, sft, data_per_streamline=sft.data_per_streamline)
```
Comment (CHrlS98) [inline scilpy/tractograms/streamline_and_mask_operations.py:132]: There is no chance that the output list of streamlines contains less streamlines than the input sft? In what case copying `data_per_streamline` will not work.

Comment (EmmaRenauld) [reply scilpy/tractograms/streamline_and_mask_operations.py]: Good question... I managed it and added a logging.info to supervise this.

**Review by arnaudbore — APPROVED** [review]
GTG

**Pull request merged**